api-gateway/src/main/java/com/petclinic/bffapigateway/domainclientlayer/AuthServiceClient.java

```diff
@@ -121,16 +121,27 @@ public Mono<OwnerResponseDTO> createUser (Mono<Register> model) {
 //                .bodyToMono(UserDetails.class);
 //    }
 
-    public Mono<UserDetails> verifyUser(final String token) {
+    public Mono<ResponseEntity<UserDetails>> verifyUser(final String token) {
+
         return webClientBuilder.build()
                 .get()
                 .uri(authServiceUrl + "/users/verification/{token}", token)
                 .retrieve()
-                .onStatus(HttpStatusCode::is4xxClientError,
-                        n -> rethrower.rethrow(n,
+                .onStatus(
+                        HttpStatusCode::is4xxClientError,
+                        n -> rethrower.rethrow(
+                                n,
                                 x -> new GenericHttpException(x.get("message").toString(), BAD_REQUEST))
```

Comment:
Why did you put BAD_REQUEST, iifc you can but n.getStatusCode ?

Reply:
It's a method used on a regular expression object to check if the string I'm giving it matches the regular expression pattern I am specifying. In this case, it is checking to see if the password matches the regex.

Reply:
Done